Xamarin.Forms.Core/FontRegistrar.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Xamarin.Forms.Internals
+{
+	public static class FontRegistrar
+	{
+		internal static readonly Dictionary<string, (ExportFontAttribute attribute, Assembly assembly)> EmbeddedFonts = new Dictionary<string, (ExportFontAttribute attribute, Assembly assembly)>();
+
+		public static void Register(ExportFontAttribute fontAttribute, Assembly assembly)
+		{
+			EmbeddedFonts[fontAttribute.FontFileName] = (fontAttribute, assembly);
+		}
+
+		//TODO: Investigate making this Async
+		public static (bool hasFont, string fontPath) HasFont(string font)
+		{
+			try
+			{
+				if (!EmbeddedFonts.TryGetValue(font, out var foundFont))
+				{
+					return (false, null);
+				}
+
+				var fontStream = GetEmbeddedResourceStream(foundFont.assembly, font);
+
+				var type = Registrar.Registered.GetHandlerType(typeof(EmbeddedFont));
+				var fontHandler = (IEmbeddedFontLoader)Activator.CreateInstance(type);
```

Comment:
was noticing you're using the Activator here... for consistency I believe this should be using the DependencyResolver/DependencyService in Xamarin.Forms... also if someone for some insane reason decided to provide their own IEmbeddedFontLoader this would give them the ability to use DI...

Reply:
I am not sure what benefit this would give us, Right now, it gets the embedded font loader from the registrar. So you can still replace it if you want.

Reply:
well I'm the first to admit I would have to ask what on earth someone is doing replacing the FontLoader... that said I just see this as a consistency issue. Anything other types that Xamarin.Forms may require can at the dev's choosing be generated by a DI container... this would explicitly prevent that.

Reply:
I am still wondering why this is considered an inconsistency issue. When you retrieve something from the Registrar, we always created it using activator. I am wondering when/where that changed? The the registrar is DI.. You can already swap things out whenever you want. You can register your own FontLoader, then it will load yours. Still failing to see an issue.

Reply:
The consistency has to do with the refactoring that was done around the DependencyResolver to add support for those who want their DI Container to be the source of truth for resolving types. The intent was that the only place in the code base that Activator.CreateType should called should be in the DependencyResolver itself for the 99% of the time where people aren't using the hook.